pkg/kubelet/dockertools/docker_network.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockertools
+
+import (
+	"fmt"
+	"sync"
+
+	dockertypes "github.com/docker/engine-api/types"
+	"github.com/golang/glog"
+
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/kubelet/network"
+)
+
+type dockerNetwork struct {
+	// Network plugin.
+	networkPlugin network.NetworkPlugin
+
+	// Pod network operations lock
+	mu sync.Mutex
```

Comment:
This is really a big lock for all pods and I wonder how much it may slow down launching/deleting pods. Why not make every pod holds its own lock?